Content.Server/GameObjects/Components/Nutrition/StomachComponent.cs

```diff
@@ -0,0 +1,117 @@
+using System;
+using Content.Shared.Chemistry;
+using Content.Shared.GameObjects.Components.Nutrition;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Serialization;
+using Robust.Shared.ViewVariables;
+
+namespace Content.Server.GameObjects.Components.Nutrition
+{
+    [RegisterComponent]
+    public class StomachComponent : SharedStomachComponent
+    {
+        // Essentially every time it ticks it'll pull out the MetabolisationAmount of reagents and process them.
+        // Generic food goes under "nutriment" like SS13
+        // There's also separate hunger and thirst components which means you can have a stomach
+        // but not require food / water.
+        public static readonly int NutrimentFactor = 30;
+        public static readonly int HydrationFactor = 30;
+        public static readonly int MetabolisationAmount = 5;
+
+        public Solution StomachContents => _stomachContents;
+        [ViewVariables]
+        private Solution _stomachContents = new Solution();
+        public int MaxVolume => _maxVolume;
+        [ViewVariables]
+        private int _maxVolume;
+        public float MetaboliseDelay => _metaboliseDelay;
+        [ViewVariables]
+        private float _metaboliseDelay; // How long between metabolisation for 5 units
+
+
+        private float _metabolisationCounter = 0.0f;
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+            serializer.DataField(ref _maxVolume, "max_volume", 20);
+            serializer.DataField(ref _metaboliseDelay, "metabolise_delay", 6.0f);
+        }
+
+        public bool TryTransferSolution(Solution solution)
+        {
+            // TODO: For now no partial transfers. Potentially change by design
+            if (solution.TotalVolume + StomachContents.TotalVolume > MaxVolume)
+            {
+                return false;
+            }
+            StomachContents.AddSolution(solution);
+            return true;
+        }
+
+        /// <summary>
+        /// This is where the magic happens. Make people throw up, increase nutrition, whatever
+        /// </summary>
+        /// <param name="solution"></param>
+        public void React(Solution solution)
+        {
+            // TODO: Optimise further?
+            var hungerUpdate = 0;
+            var thirstUpdate = 0;
+            foreach (var reagent in solution.Contents)
+            {
+                switch (reagent.ReagentId)
+                {
+                    case "chem.Nutriment":
+                        hungerUpdate++;
+                        break;
+                    case "chem.H2O":
+                        thirstUpdate++;
+                        break;
+                    case "chem.Alcohol":
+                        thirstUpdate++;
+                        break;
```

Comment:
This is something that can be done in a later PR so feel free to ignore it. Just want to make note of it. Here I think it'd be better if we had an interface `IMetabolizable` which would define the metabolization behavior of a reagent. Then each reagent would specify their metabolism type in their prototype. The benefit of this is that now the `StomachComponent` could metabolize all ingested reagents, and not just the ones preset here. Took the idea from [this post](https://github.com/space-wizards/space-station-14/issues/170#issuecomment-481835623).

Reply:
I might leave it for now but I'll definitely add a comment about it as long term it's better, cheers mate.

Reply:
Added the comment and linked to the post.